src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.BasicBlockAnalysisData.cs

```diff
@@ -112,35 +112,48 @@ public bool Equals(BasicBlockAnalysisData other)
                     return false;
                 }
 
-                var uniqueSymbols = PooledHashSet<ISymbol>.GetInstance();
-                try
+                using var _ = PooledHashSet<ISymbol>.GetInstance(out var uniqueSymbols);
+
+                // Check if both _reachingWrites maps have same set of keys.
+#if NET
+                uniqueSymbols.EnsureCapacity(Math.Max(_reachingWrites.Keys.Count, other._reachingWrites.Keys.Count));
+#endif
+
+                uniqueSymbols.AddRange(_reachingWrites.Keys);
+                uniqueSymbols.AddRange(other._reachingWrites.Keys);
```

Comment:
>uniqueSymbols.AddRange(other._reachingWrites.Keys); [](http://example.com/codeflow?start=16&length=51)

Couldn't this just use Overlaps to see if they share any keys?

Reply:
i'll look to see :)

Reply:
Going to sign off per our offline conversation. If Overlaps wouldn't suffer from the same allocation issue, then it seems like uniqueSymbols isn't needed.